### PR TITLE
[fix] Change IoTConsensusService and PipeConsensusService from async  to sync 

### PR DESCRIPTION
**Pull request by 133tosakarin**
Due to async service have some problems, we just change async method to sync.
The reason why we need change async service to sync can see https://github.com/apache/thrift/pull/3010

In this pr,  we modify IotConsensusRPCServiceProcessor and PipeConsensusRPCServiceProcessor interface  from AsyncIface to IFace so that we can use sync service.  

For async methods, we need a future as arguments and return this future with onCompete, but sync method just return the resp.

### CODE REVIEW

File: iotdb-core/consensus/src/main/java/org/apache/iotdb/consensus/iot/service/IoTConsensusRPCService.java [hunk 1]
```diff
@@ -60,14 +59,21 @@ public void initAsyncedServiceImpl(Object iotConsensusRPCServiceProcessor) {
     super.initAsyncedServiceImpl(this.iotConsensusRPCServiceProcessor);
```
[the remark this answers is not in the document]
Comment (133tosakarin) [reply iotdb-core/consensus/src/main/java/org/apache/iotdb/consensus/iot/service/IoTConsensusRPCService.java]: this line has been removed

File: iotdb-core/consensus/src/main/java/org/apache/iotdb/consensus/iot/service/IoTConsensusRPCServiceProcessor.java [hunk 2]
```diff
@@ -74,98 +73,87 @@ public IoTConsensusRPCServiceProcessor(IoTConsensus consensus) {
...
       status.setMessage(message);
-      resultHandler.onComplete(new TInactivatePeerRes(status));
-      return;
+      return new TInactivatePeerRes(status);
     }
     impl.setActive(false);
-    resultHandler.onComplete(
-        new TInactivatePeerRes(new TSStatus(TSStatusCode.SUCCESS_STATUS.getStatusCode())));
+    TInactivatePeerRes res = null;
+    res = new TInactivatePeerRes(new TSStatus(TSStatusCode.SUCCESS_STATUS.getStatusCode()));
```
[the remark this answers is not in the document]
Comment (133tosakarin) [reply iotdb-core/consensus/src/main/java/org/apache/iotdb/consensus/iot/service/IoTConsensusRPCServiceProcessor.java:148]: :)

File: iotdb-core/consensus/src/main/java/org/apache/iotdb/consensus/pipe/service/PipeConsensusRPCServiceProcessor.java [hunk 3]
```diff
@@ -61,28 +60,21 @@ public PipeConsensusRPCServiceProcessor(
...
-    try {
-      TPipeConsensusTransferResp resp = config.getConsensusPipeReceiver().receive(req);
-      // we need to call onComplete by hand
-      resultHandler.onComplete(resp);
-    } catch (Exception e) {
-      resultHandler.onError(e);
-    }
+  public TPipeConsensusTransferResp pipeConsensusTransfer(TPipeConsensusTransferReq req) {
+    TPipeConsensusTransferResp resp = config.getConsensusPipeReceiver().receive(req);
+    // we need to call onComplete by hand
```
[the remark this answers is not in the document]
Comment (133tosakarin) [reply iotdb-core/consensus/src/main/java/org/apache/iotdb/consensus/pipe/service/PipeConsensusRPCServiceProcessor.java:65]: :)

**Review by OneSizeFitsQuorum — APPROVED** [review]
LGTM

**Pull request merged**